tools/build_api.py

```diff
@@ -1357,4 +1357,4 @@ def merge_build_data(filename, toolchain_report, app_type):
                     if 'type' not in build[0]:
                         build[0]['type'] = app_type
                     build_data['builds'].insert(0, build[0])
-    dump(build_data, open(filename, "wb"), indent=4, separators=(',', ': '))
+    dump(build_data, open(filename, "w"), indent=4, separators=(',', ': '))
```

Comment:
does this close file handle after dump ?

Reply:
Echoing @jupe's comment about using `with open()`

Reply:
Ah, sorry. Typed too quickly. @jupe this is ok because the file handle is only opened within the context of the function call. Once the call is completed, the reference is garbage collected.

I agree that your suggestion is more pythonic and makes sure to close the file, but considering this is a fix and this is at the end of the file, I'm ok with this going in asis.

Reply:
It is true that when there is no references to file handle it will be closed automatically . But that is not the case when exceptions happens because then reference to file handle is stored to traceback object - part of stack trace and file will not be closed.. That could happen in some corner case - even unlikely. Anyway, it would be more safe to use `with` statement just in case..

Reply:
Interesting. Hadn't thought about what happens in the case of an exception.

I'm fine with seeing the change be made if @OPpuolitaival wants to make it. I was also under the impression that it would be good to get this in sooner rather than later, and since this seemed like a small change/improvement, I didn't want to hold up the PR.